File: python_sdk/src/reality_capture/specifications/clearance.py
from pydantic import BaseModel, Field
from typing import Optional
from enum import Enum


class ClearanceInputs(BaseModel):
    model_3d: str = Field(alias="model3d", description="Reality data id of a point cloud.")
    clearance_footprint: str = Field(alias="clearanceFootprint", description="Reality data id of building footprints.")


class ClearanceOutputs(BaseModel):
    ovf_points: Optional[str] = Field(None, alias="ovfPoints", description="Reality data id of OVF Clearance Points")
    ovf_lines: Optional[str] = Field(None, alias="ovfLines", description="Reality data id of OVF Clearance Lines")
    ovf_areas: Optional[str] = Field(None, alias="ovfAreas", description="Reality data id of OVF Clearance Areas")


class ClearanceOutputsCreate(Enum):
    OVF_POINTS = "ovfPoints"
    OVF_LINES = "ovfLines"
    OVF_AREAS = "ovfAreas"


class ClearanceSpecificationsCreate(BaseModel):
    inputs: ClearanceInputs = Field(description="Inputs")
    outputs: list[ClearanceOutputsCreate] = Field(description="Outputs")


class ClearanceSpecifications(BaseModel):
    inputs: ClearanceInputs = Field(description="Inputs")
    outputs: ClearanceOutputs = Field(description="Outputs")

File: typescript/packages/reality-capture/src/specifications/clearance.ts
import { z } from "zod";

export const ClearanceInputsSchema = z.object({
  model3d: z.string().describe("Reality data id of a point cloud"),
  clearanceFootprint: z.string().describe("Reality data id of Cbuilding footprints."),
});
export type ClearanceInputs = z.infer<typeof ClearanceInputsSchema>;

export const ClearanceOutputsSchema = z.object({
  ovfPoints: z.string()
    .describe("Reality data id of OVF Clearance Points")
    .optional(),
  ovfLines: z.string()
    .describe("Reality data id of OVF Clearance Lines")
    .optional(),
  ovfAreas: z.string()
    .describe("Reality data id of OVF Clearance Areas")
    .optional(),
});
export type ClearanceOutputs = z.infer<typeof ClearanceOutputsSchema>;

export enum ClearanceOutputsCreate {
    OVF_POINTS = "ovfPoints",
    OVF_LINES = "ovfLines",
    OVF_AREAS = "ovfAreas"
}

export const ClearanceSpecificationsCreateSchema = z.object({
  inputs: ClearanceInputsSchema.describe("Inputs"),
  outputs: z.array(z.nativeEnum(ClearanceOutputsCreate)).describe("Outputs"),
});
export type ClearanceSpecificationsCreate = z.infer<typeof ClearanceSpecificationsCreateSchema>;

export const ClearanceSpecificationsSchema = z.object({
  inputs: ClearanceInputsSchema.describe("Inputs"),
  outputs: ClearanceOutputsSchema.describe("Outputs"),
});
export type ClearanceSpecifications = z.infer<typeof ClearanceSpecificationsSchema>;